Sources/AppStoreConnectCLI/Commands/Devices/HTTPClient+DeviceId.swift

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import AppStoreConnect_Swift_SDK
+import Foundation
+import Combine
+
+extension HTTPClient {
+
+    enum DeviceIDError: Error, LocalizedError {
+        case notUnique(String)
+
+        var failureReason: String? {
+            switch self {
+            case .notUnique(let identifier):
+                return "'\(identifier)' is not a unique Device UDID."
+            }
+        }
+    }
+
+    /// Find the opaque internal identifier for a Deviec  matching `udid`. Use this for reading, modifying and deleting Device resources.
+    ///
+    /// - parameter udid: The  device UDID string.
+    /// - returns: The App Store Connect API resource identifier for the Device UDID.
+    func deviceUDIDResourceId(matching udid: String) throws -> AnyPublisher<String, Error> {
+        let request = APIEndpoint.listDevices(
+            filter: [
+                .udid([udid])
+            ]
+        )
+
+        return self.request(request)
+            .map { $0.data.filter { $0.attributes.udid == udid } }
+            .tryMap { response -> String in
+                guard response.count == 1 else {
```

Comment:
This will throw `notUnique` when nothing matches
Should we throw a different error or no error at all?

Reply:
I'll deal with this later in #77